### ISSUE TITLE
Duplicate Devices

### ISSUE THREAD

**Issue by patryan-art** [bwssytems/ha-bridge#1180]
Just to confirm, for I think I have read every post/issue on this. The problem still exists and there is no way to stop it? Problem continues to show up regardless of which Alexa device/generation you have on the network? 
Is this an Alexa issue?

Thank You

**Comment by canucitnow** [comment]
Same issues here. I take it there is no fix?

**Comment by bwssytems** [comment]
This is the same as #979

**Comment by bwssytems** [comment]
This happens by multiple ways. First is using multiple echos, the second is that the ha-bridge installation system may be providing virtual addresses that respond on each address. The second item can be controlled by fixing the ha-bridge to a specific "UPNP IP Address" in the bridge control tab.

**Comment by canucitnow** [comment]
Thank You

**Comment by patryan-art** [comment]
What/how exactly do I do for this: "by fixing the ha-bridge to a specific UPNP IP Address" in the bridge control tab". 

I tried reading more info but could not find anything exactly on what to do.
Thank You

**Comment by bwssytems** [comment]
@patryan-art You need to set the"UPNP IP Address" in the bridge control tab to the IP that your system is communicating on your local network. Google: how do I find my local ip address

**Comment by patryan-art** [comment]
I have the UPNP set up properly but continue to get duplicates.

**Comment by CPM-luvr** [comment]
My UPNP address has been set from day 1 several years ago and the duplication problem still appears, so that is not a solution.

**Comment by cylon147** [comment]
This i for All who are going through Duplicate Devices. 

I have done Extensive testing and Found that if you do discover from the Alexa Web interface it will find the Duplicate commands because when doing Discovery from the Browser it's sending some catched version of Devices which you have deleted Previously.

To fix this Problem Just do manual Deleting of all the Devices one by one from the Alexa Mobile app and Do Discovery from there.

Again I will repeat **Do not Use Alexa Web Interface.**

**Comment by superczar** [comment]
Adding to what @cylon147 said..
Here is what I have observed so far after at least a couple of years of struggling with this issue.

If you see duplicate devices :

1) No need to do manual delete 1 by 1 ...just do a forget all from the alexa browser portal (alexa.amazon.in for me, change as appropriate)

2) close your browser

3) Switch off all alexa devices except just one (very important), if you leave more than 1 on, you will almost certainly get a cached copy of dupe devices. The single alexa device should preferably be a gen 1 model (I use a gen 1 echo for rediscovery) but may work with other devices too

4) Run discovery by asking alexa. Post discovery, Verify via that all devices have been discovered with no dupes via the alexa browser portal

5) This is probably important although could be a red herring. I believe the sudden occurrence of dupes is triggered through a periodic discovery run by the mobile app. While it (probably)  won’t affect users with a single alexa device, it will affect those with multiple devices

To summarize, this is what has enabled me to run for 3+ months without dupes appearing any longer..

A) Delete alexa app from any logged in mobile devices and do not reinstall ( to avoid recurrence), run forget all from browser

B) Leave only 1 (preferably old gen) device on for discovery - Once you get all your devices in order, switch. the other devices back on

C) if you need to add any new devices/ re-run discovery - Repeat B

**Comment by AndreasGaus** [comment]
Superczar, you are my hero!
The workaround having only one old Echo device powered on when searching for new devices worked for me (see details in https://github.com/bwssytems/ha-bridge/issues/1216).
